torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -417,6 +417,12 @@ PickleOpCode Unpickler::readInstruction() {
           /*resizable=*/false); // NB: we didn't set any allocator for the
                                 // tensor
       auto options = at::CPU(type).options();
+
+      if (use_storage_device_) {
+        options = options.device(storage.device());
+        device = storage.device();
```

Comment:
This means that we will still do the `tensor = tensor.to(device)` call later down, though, right? Are we sure this is a no-op if `tensor` is already on `device`?

Reply:
Yep, it should be a no-op IIUC. I can make it more explicit by avoid that `.to()` to make it safer.